src/refdb_fs.c

```diff
@@ -1771,6 +1771,12 @@ static int reflog_append(refdb_fs_backend *backend, const git_reference *ref, co
 		goto cleanup;
 	}
 
+	if (git_path_isdir(git_buf_cstr(&path)) &&
```

Comment:
It might be nice to comment why this bit of logic is here (for example - that this is here to handle cases where a previously deeper branch was removed, but the directory structure remains...)